framework-plugins/lisk-framework-http-api-plugin/src/http_api_plugin.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { Server } from 'http';
+import { BasePlugin, ModuleInfo } from 'lisk-framework';
+import { objects } from '@liskhq/lisk-utils';
+import type {
+	BaseChannel,
+	EventsArray,
+	ActionsDefinition,
+} from 'lisk-framework';
+import * as express from 'express';
+import type { Express } from 'express';
+import * as cors from 'cors';
+import * as rateLimit from 'express-rate-limit';
+import * as controllers from './controllers';
+import * as middlewares from './middlewares';
+import * as config from './defaults';
+import { Options } from './types';
+
+export class HTTPAPIPlugin extends BasePlugin {
+	private _server!: Server;
+	private _app!: Express;
+	private _channel!: BaseChannel;
+
+	// eslint-disable-next-line @typescript-eslint/class-literal-property-style
+	public static get alias(): string {
+		return 'http_api';
+	}
+
+	// eslint-disable-next-line @typescript-eslint/class-literal-property-style
+	public static get info(): ModuleInfo {
+		return {
+			author: '@liskhq',
+			version: '0.1.0',
+			name: '@liskhq/lisk-framework-http-api-plugin',
+		};
```

Comment:
i was trying to do that, but it cannot import the package json as it is. and it needs to read the file using `fs`.
I wasnt sure if this is required information or not, ie in the framework we have this, but never updated and used

Reply:
ya.. either we remove everywhere or use from package.. as it will be maintained consistently.

Reply:
I think its very useful information in terms of plugin that is aware of its context by itself. And yes preferred to read through the package.json file to make sure it never miss-match actual values.

Reply:
We used to have a similar approach to the blueprint that we build last year. 

https://github.com/LiskHQ/lisk-modular/blob/2e761eaeb93c4f30110e47829f9189dd02e10d7b/modules/lisk-core-chain/src/index.js#L8

Reply:
it cannot read the file directly as import or require since package.json is outside of the build directory, and location of package json cannot be modified.
It needs to use `fs`, but using `fs` in the static method, im not sure it's a good approach

Reply:
Updated to use `require` instead